chainer/functions/math/basic_math.py

```diff
@@ -368,35 +392,55 @@ def forward_cpu(self, inputs):
         self.retain_inputs((0, 1, 2))
         x0, x1, gy = inputs
         gx0 = utils.force_array(gy / x1)
-        return gx0, utils.force_array(-gx0 * x0 / x1)
+        gx1 = utils.force_array(-gx0 * x0 / x1)
+        return utils.sum_to(gx0, x0.shape), utils.sum_to(gx1, x1.shape)
 
     def forward_gpu(self, inputs):
         self.retain_inputs((0, 1, 2))
         x0, x1, gy = inputs
-        return cuda.elementwise(
+        gx0, gx1 = cuda.elementwise(
             'T x0, T x1, T gy',
             'T gx0, T gx1',
             '''
                gx0 = gy / x1;
                gx1 = -gx0 * x0 / x1;
             ''', 'div_bwd')(x0, x1, gy)
+        return utils.sum_to(gx0, x0.shape), utils.sum_to(gx1, x1.shape)
 
     def backward(self, indexes, grad_outputs):
         x0, x1, gy = self.get_retained_inputs()
         ggx0, ggx1 = grad_outputs
 
         ret = []
         x1_square = x1 * x1
+
         if 0 in indexes:
-            gx0 = -ggx1 * gy / x1_square
-            ret.append(gx0)
+            if ggx1 is None:
+                ret.append(None)
+            else:
+                gx0 = -ggx1 * gy / x1_square
+                ret.append(chainer.functions.sum_to(gx0, x0.shape))
+
         if 1 in indexes:
-            gx1 = -ggx0 * gy / x1_square + \
-                ggx1 * 2 * gy * x0 / (x1_square * x1)
-            ret.append(gx1)
+            gx1 = None if ggx0 is None else -ggx0 * gy / x1_square
+            gx1_1 = (None if ggx1 is None else
+                     ggx1 * 2 * gy * x0 * (x1_square * x1))
```

Comment:
`ggx1 * 2 * gy * x0 * (x1_square * x1)` should be
`ggx1 * 2 * gy * x0 / (x1_square * x1)`

Reply:
Thanks, I fixed it.